src/main/java/com/puppycrawl/tools/checkstyle/xpath/iterators/PrecedingIterator.java

```diff
@@ -0,0 +1,88 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath.iterators;
+
+import net.sf.saxon.om.AxisInfo;
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.tree.iter.AxisIterator;
+
+/**
+ * Recursive-free implementation of the preceding axis iterator.
+ */
+public class PrecedingIterator implements AxisIterator {
+    /**
+     * Ancestor axis iterator.
+     */
+    private final AxisIterator ancestorEnum;
+    /**
+     * Preceding-sibling axis iterator.
+     */
+    private AxisIterator siblingEnum;
```

Comment:
I think it makes more sense to rename this to `previousSiblingEnum`. All I see is `sibling.next` and it makes me think we are going forward, when we are actually going backways by the class name.

Reply:
done

Reply:
reverted back, otherwise it won't pass test coverage

Reply:
How is a name connected to test coverage?

Reply:
and yes, this one was done